src/App.tsx

```diff
@@ -1,41 +1,72 @@
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [selectedUserId, setSelectedUserId] = useState<number>(0);
+  const [selectedPostId, setSelectedPostId] = useState<number | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
 
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
+  const loadPosts = async () => {
+    setPosts(
+      await getUserPosts(selectedUserId),
+    );
+  };
 
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
+  useEffect(() => {
+    setIsLoading(true);
+    loadPosts().then(() => setIsLoading(false));
+  }, [selectedUserId]);
 
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+  return (
+    <div className="App">
+      <header className="App__header">
+        <label>
+          Select a user: &nbsp;
+          <select
+            className="App__user-selector"
+            value={selectedUserId}
+            onChange={(event) => setSelectedUserId(+event.target.value)}
+          >
+            <option value="0">All users</option>
+            <option value="1">Leanne Graham</option>
+            <option value="2">Ervin Howell</option>
+            <option value="3">Clementine Bauch</option>
+            <option value="4">Patricia Lebsack</option>
+            <option value="5">Chelsey Dietrich</option>
+            <option value="6">Mrs. Dennis Schulist</option>
+            <option value="7">Kurtis Weissnat</option>
+            <option value="8">Nicholas Runolfsdottir V</option>
+            <option value="9">Glenna Reichert</option>
+            <option value="10">Leanne Graham</option>
+          </select>
+        </label>
+      </header>
+
+      <main className="App__main">
+        <div className="App__sidebar">
+          <PostsList
+            posts={posts}
+            isLoading={isLoading}
+            selectedPostId={selectedPostId}
+            onPostSelect={setSelectedPostId}
+          />
+        </div>
+
+        {selectedPostId ? (
+          <div className="App__content">
+            <PostDetails
+              selectedPostId={selectedPostId}
+            />
+          </div>
+        ) : ('')}
```

Comment:
```suggestion
        {selectedPostId && (
          <div className="App__content">
            <PostDetails
              selectedPostId={selectedPostId}
            />
          </div>
        )}
```
you don't need empty string

Reply:
in this case, maybe you can help me with this bug. If I go with && approach, after selecting and de-selecting the post, there is a 0 that appears in the HTML file. The ? () : ('') approach fixed it. Again, it was deliberate.
![Screenshot 2022-06-30 at 23 11 07](https://user-images.githubusercontent.com/57355180/176769002-276e14ee-9ef9-4f30-ae1b-60763ba9003e.jpg)
![Screenshot 2022-06-30 at 23 11 40](https://user-images.githubusercontent.com/57355180/176769097-242cfea1-0436-4f22-8d6b-7cf4ea058be6.jpg)
![Screenshot 2022-06-30 at 23 12 05](https://user-images.githubusercontent.com/57355180/176769221-8582eda7-3e44-4b6f-b290-e0cfbcbeae35.jpg)

not making changes as of yet to avoid this bug

Reply:
you can do like this to avoid this bug `Boolean(selectedPostId)`